schema/query/resolver.go

```diff
@@ -0,0 +1,143 @@
+package query
+
+import (
+	"context"
+	"sync"
+
+	"github.com/rs/rest-layer/schema"
+)
+
+type referenceResponseHandler func(payloads []map[string]interface{}, validator schema.Validator) error
+
+type referenceBatchResolver struct {
+	mu       sync.Mutex
+	requests []referenceRequest
+}
+
+func (rbr *referenceBatchResolver) request(resourcePath string, q *Query, handler referenceResponseHandler) {
+	rbr.mu.Lock()
```

Comment:
Only one `request` might run at the time per rbr instance - is that intentional?